packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx

```diff
@@ -492,6 +654,18 @@ function WorklistDataGrid(props) {
       default:
         return;
     }
+    if (activateMultiRowSelection) {
+      // Handle the row selection case.
+      if (nextRow > rows.length) {
+        nextRow = rows.length;
+      } else if (nextRow <= 0) {
+        nextRow = 1; // Only non-header rows can be selected.
+      }
+      rowSelection.current.isRowSelectionModeToggledByGrid = !hasSelectableRows;
+      selectMultipleRows(nextRow, rowSelection.current.rangeUsingFocusedRow);
+      rowSelection.current.rangeUsingFocusedRow.previousSelectionEndRow = nextRow;
+    }
```

Comment:
This event does bubble.  What happens if the users is in an input or text area element or contenteditable div?  We would not want to execute this new logic on each key press, right?

Reply:
The events will still fire. As discussed, we would have to devise a solution to handle events that originate from consumer components on the grid. We will also have to figure what should happen when the Worklist Data Grid keyboard shortcuts are triggered from within consumer components on the grid. For example, what happens when Ctrl+A is pressed from inside the input textbox within the grid. We may have to open another story if necessary.

Reply:
Ctrl+A should select all the text inside the input textbox in that scenario.